tests/unit/api/objectPutLegalHold.js

```diff
@@ -0,0 +1,102 @@
+const assert = require('assert');
+
+const { bucketPut } = require('../../../lib/api/bucketPut');
+const objectPut = require('../../../lib/api/objectPut');
+const objectPutLegalHold = require('../../../lib/api/objectPutLegalHold');
+const { cleanup, DummyRequestLogger, makeAuthInfo } = require('../helpers');
+const metadata = require('../../../lib/metadata/wrapper');
+const DummyRequest = require('../DummyRequest');
+
+const log = new DummyRequestLogger();
+
+const authInfo = makeAuthInfo('accessKey1');
+const namespace = 'default';
+const bucketName = 'bucketname';
+const objectName = 'objectName';
+const postBody = Buffer.from('body', 'utf8');
+
+const putBucketRequest = {
+    bucketName,
+    headers: { host: `${bucketName}.s3.amazonaws.com` },
+    url: '/',
+};
+
+const putObjectRequest = new DummyRequest({
+    bucketName,
+    namespace,
+    objectKey: objectName,
+    headers: {},
+    url: `/${bucketName}/${objectName}`,
+}, postBody);
+
+const objectLegalHoldXml = status => '<LegalHold ' +
```

Comment:
I made this change after your comment but just reverted back after finding out whitespace errors with XMLs due to backticks adding line breaks and extra whitespace (an alternative could be writing it on one line but I thought that would probably be less ideal).

<img width="791" alt="objectLegalHold_js_—_cloudserver" src="https://user-images.githubusercontent.com/62902974/83556349-90511680-a4c4-11ea-82ab-abd46ae0f1c1.png">

Reply:
Backticks will take the string verbatim including newlines and spaces. It is valid XML but your test does not include the newlines or spaces so is causing a not equal assertion. This is one of many reasons I think string to string assertions are cumbersome but meh. Building an XML file or object from actual strings is generally not ideal IMO. It is like you generally don't build JSON from strings and instead use an object and stringify it. In doing so you avoid string specific errors, especially around ordering and whitespace. Regardless, valid XML is valid XML and I have no concern with your string here. I would recommend finding a way to generate XML from a more static object such as an object or JSON like structure for sanity purposes.

Reply:
Yeah I thought of adjusting the test instead as an alternative so they would match but couldn't be sure if line breaks or whitespace would cause any unforeseen issues.

I cannot agree more about building XML from actual strings, like 110% :)
I think writing API utils to generate XML from JSON (and vice-versa) is a good idea. It could be really beneficial to have a standardized method for it to be utilized across the other APIs too. Just having a method to generate the XML from the object or JSON would be good enough for a start I think. Then we can start refactoring the existing APIs gradually as well. I'll create a ticket for it so it wouldn't get missed out or forgotten.

Reply:
Might want to search NPM for an existing package, I can't imagine it doesn't exist already.